### ISSUE TITLE
Postal see but ignores viruses found by ClamAV

### ISSUE THREAD

**Issue by dragoangel** [postalserver/postal#1293]
Hi,

I set clamav and tested setup with publicly known virus samples.
I tested with Eicar virus and it failed to block it, okay I found real viruses at https://dasmalwerk.eu/ 
I downloaded and extracted Trojan.GenericKD.40427213 sample virus from zip file (as it secured by password) before sending.
Clamav detects it as: https://www.virustotal.com/gui/file/7682b842ed75b69e23c5deecf05a45ee79c723d98cfb6746380d748145bfc1af/detection and:

1. send it as postal user
2. send it as external mail server to postal mx which them routed to external email 👎 

In both cases email was been delivered back and forth.
In clamav logs after sending email I see:
`instream(127.0.0.1@40612): Win.Dropper.Fareit-9802767-0(bf340b3ff326c...`

In postal I see:
`Threat: Win.Dropper.Fareit-9802767-0`

To note I enabled `Outbound spam threshold to 5.0` and virus symbol wasn't been added to spam checks scores.

**Comment by willpower232** [comment]
Have you enabled a clamav integration? It is different to spamassassin.

Unfortunately it is not documented in the wiki but you can install clamav and provide details in the config like this
https://github.com/postalhq/postal/blob/2595481b266199d8e29347dea2764f2c10202805/config/postal.defaults.yml#L121-L124

**Comment by dragoangel** [comment]
@willpower232 hi,
yes of course, otherwise Postal UI will not show me: `Threat Win.Dropper.Fareit-9802767-0` 😄

**Comment by willpower232** [comment]
haha not reading well enough today!

Yes this doesn't seem to be implemented but I'm not sure why.

Postal does include a header in the message https://github.com/postalhq/postal/blob/2595481b266199d8e29347dea2764f2c10202805/app/jobs/unqueue_message_job.rb#L132

but I'm not sure if that would do any good, it would be interesting to have some clarification from @adamcooke or @catphish

**Comment by dragoangel** [comment]
As Postal not handle viruses in combination of antispam solution: there must be an option in Postal to not allow receive or send viruses separately. Of course in general I think nobody want to get or send such stuff at all...

**Comment by dragoangel** [comment]
@willpower232 update:

1. On outgoing email header X-Postal-Threat isn't set at all.
2. On incomming email header is `X-Postal-Threat: no` while actually `Threat` is found.

So even if I add something like this:
```
              #
              # If this message has a X-Postal-Threat: yes
              #
              if queued_message.message.threat == 1
                log "#{log_prefix} Message has a virus. Hard failing."
                queued_message.message.create_delivery('HardFail', :details => "Message's has a virus.")
                queued_message.destroy
                next
              end
```

It will not fix anything

**Comment by dragoangel** [comment]
Also question about virus file results caching: from for example rspamd I know that it done in background and then results is caches in redis cache for some time, so mail with same viruses non scanned twise, they determined by cached bad results. In general it would be cool that postal will not try to do all this by himself, but had integrated with more advanced antispam solution then spam assassin which has own capitabiluties to detect viruses and return one score for mail including score of virus symbol. The one thing rspamd use milters to integrate with mta, can postal support it?

**Comment by dragoangel** [comment]
@willpower232 looks like people tried to particulary fix issue, at least with headers.

**Comment by catphish** [comment]
The X-Postal-Threat header was not correctly set. This bug is fixed by 1e3622c49abeeb0ca8a7cc1224e84f2fc74cc938
Detected threats should also be displayed in the web interface.
Currently there is no setting to block messages with viruses though.
